### PR TITLE
Added support for trimmed mean calculation: TDIGEST.TRIMMED_MEAN

### LINKED ISSUE #386: T-Digest: enable trimmed mean calculations
Feature: return the mean of data set without the values below and above the p1 and p2 percentile respectively.

expected API:
```
TDIGEST.TRIMMED_MEAN {key} {low_cut_percentile} {high_cut_percentile}
```

Is blocked by: https://github.com/RedisBloom/t-digest-c/issues/16

### PR DESCRIPTION
**Pull request by filipecosta90**
The following PR adds support for trimmed mean calculation via the new command:
```
TDIGEST.TRIMMED_MEAN {key} {low_cut_percentile} {high_cut_percentile}
```
i.e. return the mean of the data set without the values below and above the low-cut and high-cut percentiles respectively.

Fixes #386 

Note: It implies the review of https://github.com/RedisBloom/t-digest-c/pull/22

### CODE REVIEW

File: src/rm_tdigest.c [hunk 1]
```diff
@@ -311,6 +311,50 @@ int TDigestSketch_Cdf(RedisModuleCtx *ctx, RedisModuleString **argv, int argc) {
     return REDISMODULE_OK;
 }
 
+/**
+ * Command: TDIGEST.TRIMMED_MEAN {key} {low_cut_percentile} {high_cut_percentile}
+ *
+ * Returns the trimmed mean ignoring values outside given cutoff upper and lower limits.
+ *
+ * @param ctx Context in which RedisTRIMMED_MEAN modules operate
```
Comment (ashtul) [inline src/rm_tdigest.c:319]: clean TRIMMED_MEAN

File: src/rm_tdigest.c [hunk 2]
```diff
@@ -311,6 +311,50 @@ int TDigestSketch_Cdf(RedisModuleCtx *ctx, RedisModuleString **argv, int argc) {
...
+    double high_cut_percentile = 0.0;
+    if (RedisModule_StringToDouble(argv[2], &low_cut_percentile) != REDISMODULE_OK) {
+        RedisModule_CloseKey(key);
+        return RedisModule_ReplyWithError(ctx, "ERR T-Digest: error parsing low_cut_percentile");
+    }
+    if (RedisModule_StringToDouble(argv[3], &high_cut_percentile) != REDISMODULE_OK) {
+        RedisModule_CloseKey(key);
+        return RedisModule_ReplyWithError(ctx, "ERR T-Digest: error parsing high_cut_percentile");
+    }
+    if (low_cut_percentile < 0.0 || low_cut_percentile > 1.0 || high_cut_percentile < 0.0 ||
```
Comment (ashtul) [inline src/rm_tdigest.c:346]: Add check for `low_cut_percentile > high_cut_percentile`?

File: tests/flow/test_tdigest.py [hunk 3]
```diff
@@ -357,6 +357,55 @@ def test_negative_tdigest_cdf(self):
...
+            self.assertOk(self.cmd("tdigest.add", "tdigest", x, 1.0))
+
+        self.assertAlmostEqual(
+            9.5, float(self.cmd("tdigest.trimmed_mean", "tdigest", 0.1,0.9)), 0.01
+        )
+        self.assertAlmostEqual(
+            9.5, float(self.cmd("tdigest.trimmed_mean", "tdigest", 0.0,1.0)), 0.01
+        )
+        self.assertAlmostEqual(
+            9.5, float(self.cmd("tdigest.trimmed_mean", "tdigest", 0.20,0.8)), 0.01
```
Comment (ashtul) [inline tests/flow/test_tdigest.py:373]: 0.2

**Review by ashtul — COMMENTED** [review]
Looking good.
Minor comments

File: src/rm_tdigest.c (hunk 2 above)
Comment (filipecosta90) [reply src/rm_tdigest.c:346]: Fixed in the latest push

**Review by ashtul — DISMISSED** [review]
LGTM

**Review by ashtul — APPROVED** [review]

**Review by ashtul — APPROVED** [review]

**Pull request merged**